app_dart/lib/src/request_handlers/push_gold_status_to_github.dart

```diff
@@ -0,0 +1,274 @@
+            log.debug('Notifying for triage.');
+            await _commentAndApplyGoldLabel(gitHubClient, pr, slug);
+          }
+        }
+      } else {
+        // We have seen this commit before.
+        // If checks are still running, or last update was green, do nothing.
+        if (cirrusCheckStatuses.any(cirrusInProgressStates.contains) ||
+            lastUpdate.status == GithubGoldStatusUpdate.statusCompleted)
+          return Body.empty;
+
+        // Check Gold for new status and update.
+        final String status = await _getGoldStatus(pr, log);
+        log.debug('Checks are completed, Gold reports $status status for '
+            '${pr.number} sha ${pr.head.sha}.');
+        if (lastUpdate.status != status) {
+          // The 'running' state is used when the check is waiting and when
+          // triage is needed, so if it is already 'running' we don't need to
+          // update it. The comment feature below alerts the author that the
+          // result needs to be addressed.
+          statusRequest = CreateStatus(status);
+          statusRequest.targetUrl = kTargetUrl;
+          statusRequest.context = kContext;
+          statusRequest.description = _getStatusDescription(status);
+        }
+        if (status == GithubGoldStatusUpdate.statusRunning &&
+            !await _alreadyCommented(gitHubClient, pr, slug)) {
+          log.debug('Notifying for triage.');
+          await _commentAndApplyGoldLabel(gitHubClient, pr, slug);
+        }
+      }
+
+      if (statusRequest != null) {
+        try {
+          await gitHubClient.repositories
+              .createStatus(slug, pr.head.sha, statusRequest);
+          lastUpdate.status = statusRequest.state;
+          lastUpdate.head = pr.head.sha;
+          lastUpdate.updates += 1;
+          statusUpdates.add(lastUpdate);
+        } catch (error) {
+          log.error(
+              'Failed to post status update to ${slug.fullName}#${pr.number}: $error');
+        }
+      }
+    }
+
+    final int maxEntityGroups = config.maxEntityGroups;
+    for (int i = 0; i < statusUpdates.length; i += maxEntityGroups) {
+      await datastore.db.withTransaction<void>((Transaction transaction) async {
+        transaction.queueMutations(
+            inserts: statusUpdates.skip(i).take(maxEntityGroups).toList());
+        await transaction.commit();
+      });
+    }
+    log.debug('Committed all updates');
+    return Body.empty;
+  }
+
+  /// Used to check for any tryjob results from Flutter Gold associated with a
+  /// pull request.
+  Future<String> _getGoldStatus(PullRequest pr, Logging log) async {
+    // We wait for a few seconds in case tests _just_ finished and the tryjob
+    // has not finished ingesting the results.
+    await Future<void>.delayed(const Duration(seconds: 10));
+    final Uri requestForTryjobStatus = Uri.parse(
+        'http://flutter-gold.skia.org/json/changelist/github/${pr.number}/${pr.head.sha}/untriaged');
```

Comment:
@kjlubick are there any rate concerns here?

I have tried to limit this to only request for pull requests running tryjobs (not all PRs run framework tests), only when all the other checks are completed on a commit., then if the first check isn't successful (there are changes), it will check again in three minutes. 

So, for most cases I think this will hit only once, reporting no changes and not request again unless a new commit is pushed and all the checks are complete.

Reply:
I have rate limiting and some short-lived caching on my side, so what you have described should be fine.  If not, I'll write better stuff ;)